src/main.cpp

```diff
@@ -2223,19 +2202,14 @@ void setup() {
     bPID.SetSmoothingFactor(EMA_FACTOR);
     bPID.SetMode(AUTOMATIC);
 
-    // Dallas temp sensor
-    #if TEMPSENSOR == 1
-        sensors.begin();
-        sensors.getAddress(sensorDeviceAddress, 0);
-        sensors.setResolution(sensorDeviceAddress, 10);
-        sensors.requestTemperatures();
-        temperature = sensors.getTempCByIndex(0);
-    #endif
+    if (TEMP_SENSOR == 1) {
+        tempSensor = new TempSensorDallas(PIN_TEMPSENSOR);
+    }
+    else if (TEMP_SENSOR == 2) {
+        tempSensor = new TempSensorTSIC(PIN_TEMPSENSOR);
+    }
```

Comment:
We know this at compile-time. I would directly set this in the top of the file along with the rest of the hardware. That way you don;t run the risk that this code is somehow not reached and you are operating with a `nullptr` whenever calling `tempSensor->something()` and crash the controller.

Reply:
Optimally we would not even include the Dallas sensor lib in the build, so only required code ends up on the device.